code/socialDistribution/templates/tagtemplates/modal.html

```diff
@@ -0,0 +1,36 @@
+
+{% load post %}
+<div class="modal fade" id="{{modal_id}}" tabindex="-1" aria-labelledby="{{modal_label}}" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h6 class="modal-title" id="socialDistModalTitle">{{modal_title}}</h6>
+                <button class="close" type="button" data-bs-dismiss="modal" aria-label="Close"
+                    style="
+                        background-color: transparent;
+                        border: none;
+                    "
+                >
+                    <span aria-hidden="true">&times;</span>
+                </button> 
+            </div>
+            <div class="modal-body">
+                {% if modal_type == "post" %}
+                {% post_form %}
```

Comment:
The modal will probably be used in other places within the app. Extend this if else and modify the modal.py file to reuse this modal template.